graphql/src/schema/api.rs

```diff
@@ -296,15 +308,6 @@ fn add_filter_type(
     let filter_type_name = format!("{}_filter", type_name).to_string();
     match ast::get_named_type(schema, &filter_type_name) {
         None => {
-            let input_values = field_input_values(schema, fields)?;
-
-            // Don't generate an input object with no fields, this makes the JS
-            // graphql library, which graphiql uses, very confused and graphiql
-            // is unable to load the schema. This happens for example with the
-            // definition `interface Foo { x: OtherEntity }`.
-            if input_values.is_empty() {
-                return Ok(());
-            }
```

Comment:
There is no such case anymore as far as I can tell and the example given here doesn't seem correct because it then still lets you filter by the reference entity id? Hence, I removed this bit and also simplified the logic accordingly in the other places where this was then taken care of.